### PR TITLE
Custom repository

### PR DESCRIPTION
**Pull request by efraespada**
## ISSUE
Necessity for a bit more customization 

## Description
- Enabled Places API
- Added option for implementing a custom data source
- Added option for implementing a custom autocomplete results adapter

No visual/UX or breaking changes

## Mandatory GIF
mandatory

### CODE REVIEW

File: leku/src/main/java/com/adevinta/leku/geocoder/GoogleGeocoderDataSource.kt [hunk 1]
```diff
@@ -5,97 +5,143 @@ import com.google.android.gms.maps.model.LatLng
...
+                String.format(Locale.ENGLISH, QUERY_PLACE_ID, placeId, placesApiKey)
+            )
+            when {
+                result != null -> addressBuilder.parseResult(result)
+                else -> null
+            }
+        } catch (e: JSONException) {
+            null
+        } catch (e: NetworkException) {
+            null
```
Comment (jlquintana) [inline leku/src/main/java/com/adevinta/leku/geocoder/GoogleGeocoderDataSource.kt:75]: I'm not sure you have to hide this kind of error. Maybe the client wants to handle them.

File: leku/src/main/java/com/adevinta/leku/geocoder/GoogleGeocoderDataSource.kt [hunk 2]
```diff
@@ -5,97 +5,143 @@ import com.google.android.gms.maps.model.LatLng
...
+                String.format(Locale.ENGLISH, QUERY_AUTOCOMPLETE, query, placesApiKey)
+            )
+            if (result != null) {
+                suggestions.addAll(suggestionBuilder.parseResult(result))
+            }
+            suggestions
+        } catch (e: JSONException) {
+            suggestions
+        } catch (e: NetworkException) {
+            suggestions
```
Comment (jlquintana) [inline leku/src/main/java/com/adevinta/leku/geocoder/GoogleGeocoderDataSource.kt:56]: Same here

**Review by jlquintana — APPROVED** [review]
Approved with non-blocking comments

File: leku/src/main/java/com/adevinta/leku/geocoder/GoogleGeocoderDataSource.kt (hunk 1 above)
Comment (efraespada) [reply leku/src/main/java/com/adevinta/leku/geocoder/GoogleGeocoderDataSource.kt:75]: I kept the same `Exception` treatment with new data source classes: return an empty list or `null`

**Review by jlquintana — APPROVED** [review]

**Review by ferranpons — APPROVED** [review]

**Pull request merged**